scipy/stats/_hypotests.py

```diff
@@ -389,3 +390,192 @@ def _get_wilcoxon_distr(n):
                          "statistic is not implemented for n={}".format(n))
 
     return np.array(cnt, dtype=int)
+
+
+def _pval_cvm_2samp_exact(s, nx, ny):
+    """
+    Compute the exact p-value of the Cramer-von Mises two-sample test
+    for a given value s (float) of the test statistic by enumerating
+    all possible combinations. nx and ny are the sizes of the samples.    
```

Comment:
When I first read this, I was surprised that it was combinations, not permutations. I (think I) now understand that while the null distribution can be thought of as the test statistic of all `binom(nx + ny, nx)` permutations of the data into each group, each value of the statistic would then be repeated `factorial(nx)` times, so you can get the same distribution for the sake of calculating a p-value using combinations. Is that right?

Reply:
- I think your reasoning is correct. The key point is to look at the ranks, i.e. after sorting the data, so there is no need to work with all possible permutations of the data.
- yes, the calculation is very similar to the KS test exact method and there are more efficient path counting methods comparable to the code implemented by @pvanmulbregt . I just wanted to add a simple way to compute the exact values for small samples.

Reply:
OK.